lsp/nls/src/requests/completion.rs

```diff
@@ -29,18 +29,109 @@ use crate::{
 // We follow the path by traversing a term, type or contract which represents a record
 // and stop when there is nothing else on the path
 
+struct IdentWithMeta {
+    ident: Ident,
+    ty: Types,
+    item: Option<LinearizationItem<Types>>,
+}
+
+impl From<Ident> for IdentWithMeta {
+    fn from(ident: Ident) -> Self {
+        IdentWithMeta {
+            ident,
+            ty: Types(TypeF::Dyn),
+            item: None,
+        }
+    }
+}
+
+impl From<&str> for IdentWithMeta {
+    fn from(ident: &str) -> Self {
+        IdentWithMeta {
+            ident: Ident::from(ident),
+            ty: Types(TypeF::Dyn),
+            item: None,
+        }
+    }
+}
+
+impl IdentWithMeta {
+    fn compute_detail(&self) -> String {
+        self.item
+            .as_ref()
+            .and_then(|item| {
+                item.meta.as_ref().and_then(|meta| {
+                    meta.types
+                        .as_ref()
+                        .map(|ty| ty.types.to_string())
+                        .or_else(|| meta.contracts_to_string())
+                })
+            })
+            .unwrap_or_else(|| self.ty.to_string())
+    }
+
+    fn compute_completion_item_kind(&self) -> CompletionItemKind {
+        match &self.ty {
+            ty if ty.is_function_type() => CompletionItemKind::Function,
+            _ => CompletionItemKind::Property,
+        }
+    }
+
+    fn to_lsp_completion_item(&self) -> CompletionItem {
+        /// Attach quotes to a non-ASCII string
+        fn adjust_name(name: &str) -> String {
+            if name.is_ascii() {
+                String::from(name)
+            } else {
+                format!("\"{}\"", name)
+            }
+        }
+        let doc = || {
+            let item = self.item.as_ref()?;
+            let meta = item.meta.as_ref()?;
+            let doc = meta.doc.as_ref()?;
+            let doc =
+                Documentation::String(format!("{}\nMerge Priority {}", doc.clone(), meta.priority));
```

Comment:
I think you have a way to pretty-print metadata somewhere in the code, for the REPL and `nickel query` (including a version without markdown support). I'm not saying this should be done in this PR, but maybe we could reuse the same code, in order to have a consistent way of showing metadata about values.

Reply:
Yes, I agree. I've already done a little bit in this PR: when we display metadata on LSP hover, there is a  separate pretty-printing for that, so I moved some of that to be methods of `MetaValue`